src/group_theory/coset.lean

```diff
@@ -303,6 +303,21 @@ quotient_group.eq
 lemma out_eq' (a : quotient s) : mk a.out' = a :=
 quotient.out_eq' a
 
+variables (s)
+
+/- It can be useful to write `obtain ⟨h, H⟩ := mk_out'_eq_mul ...`, and then `rw [H]` or
+  `simp_rw [H]` or `simp only [H]`. In order for `simp_rw` and `simp only` to work, this lemma is
+  stated in terms of an arbitrary `h : s`, rathern that the specific `h = g⁻¹ * (mk g).out'`. -/
+@[to_additive quotient_add_group.mk_out'_eq_mul]
+lemma mk_out'_eq_mul (g : α) : ∃ h : s, (mk g : quotient s).out' = g * h :=
+⟨⟨g⁻¹ * (mk g).out', eq'.mp (mk g).out_eq'.symm⟩, by rw [s.coe_mk, mul_inv_cancel_left]⟩
+
+variables {s}
+
+@[to_additive quotient_add_group.mk_mul_of_mem]
+lemma mk_mul_of_mem (g₁ g₂ : α) (hg₂ : g₂ ∈ s) : (mk (g₁ * g₂) : quotient s) = mk g₁ :=
```

Comment:
Is this variant also true?
```lean
lemma mk_mul_of_mem_left (g₁ g₂ : α) (hg₁ : g₁ ∈ s) : (mk (g₁ * g₂) : quotient s) = mk g₂ :=
```

Reply:
I don't think so (left cosets vs right cosets).